sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/ExpressionParserSuite.scala

```diff
@@ -179,6 +179,40 @@ class ExpressionParserSuite extends PlanTest {
       InSubquery(Seq('a), ListQuery(table("c").select('b))))
   }
 
+  test("any sub-query") {
+    // Since the `genCmp` is generated from `AstBuilder` as a lambda expression
+    // which is not equal to the original comparison expression like `EqualTo`,
+    // we need to extract it from the parsed expression.
+    def extractPredicate(expr: Expression): (Expression, Expression) => Expression = {
+      expr.asInstanceOf[AnySubquery].genCmp
+    }
+
+    val any_subquery = defaultParser.parseExpression("a < any (select b from c)")
+    compareExpressions(
```

Comment:
plz use `assertEqual`

Reply:
As I comment in https://github.com/apache/spark/pull/23783#discussion_r263353085 , we have to use `compareExpressions` to compare two `Expression`.

Reply:
How about this? https://github.com/apache/spark/commit/4d44e384fc9e1987fec89086d8d984fadd836a97

Reply:
Yes, it works. Thanks again.